apps/api/src/deployment/services/deployment/deployment.service.ts

```diff
@@ -0,0 +1,83 @@
+import { MsgCreateDeployment } from "@akashnetwork/akash-api/v1beta3";
+import { SDL } from "@akashnetwork/akashjs/build/sdl";
+import { getAkashTypeRegistry } from "@akashnetwork/akashjs/build/stargate";
+import { DirectSecp256k1HdWallet, Registry } from "@cosmjs/proto-signing";
+import { SigningStargateClient } from "@cosmjs/stargate";
+import axios from "axios";
+import { singleton } from "tsyringe";
+
+import { BillingConfigService } from "@src/billing/services/billing-config/billing-config.service";
+import { SingleDeploymentResponse } from "@src/deployment/http-schemas/deployment.schema";
+import { MessageTransmitterService } from "@src/deployment/services/message-transmitter/message-transmitter.service";
+import { RestAkasheploymentInfoResponse, RestAkashLeaseListResponse } from "@src/types/rest";
+import { apiNodeUrl, betaTypeVersion, betaTypeVersionMarket } from "@src/utils/constants";
+import { env } from "@src/utils/env";
+
+@singleton()
+export class DeploymentService {
+  constructor(
+    private readonly config: BillingConfigService,
+    private readonly messageTransmitterService: MessageTransmitterService,
+  ) { }
+
+  public async findByOwnerAndDseq(owner: string, dseq: string): Promise<SingleDeploymentResponse['data']> {
+    const response = await axios.get<RestAkasheploymentInfoResponse>(`${apiNodeUrl}/akash/deployment/${betaTypeVersion}/deployments/info`, {
+      params: {
+        "id.owner": owner,
+        "id.dseq": dseq,
+      }
+    });
+
+    if ('code' in response.data) {
+      return {
+        deployment: null,
+        leases: [],
+        escrow_account: null
+      }
+    }
+
+    const { data: { leases } } = await axios.get<RestAkashLeaseListResponse>(`${apiNodeUrl}/akash/market/${betaTypeVersionMarket}/leases/list`, {
+      params: {
+        "filters.owner": owner,
+        "filters.dseq": dseq
+      }
+    });
+
+    return {
+      deployment: response.data.deployment,
+      leases: leases.map(({ lease }) => lease),
+      escrow_account: response.data.escrow_account
+    };
+  }
+
+  public async create(sdl: SDL) {
+    const dseq = (await this.messageTransmitterService.getCurrentHeight()).toString();
+    const manifestVersion = await sdl.manifestVersion();
+    const wallet = await DirectSecp256k1HdWallet.fromMnemonic(env.GPU_BOT_WALLET_MNEMONIC, { prefix: "akash" });
+    const [account] = await wallet.getAccounts();
+    const myRegistry = new Registry([...getAkashTypeRegistry()]);
+    const client = await SigningStargateClient.connectWithSigner(this.config.get("RPC_NODE_ENDPOINT"), wallet, {
+      registry: myRegistry,
+      broadcastTimeoutMs: 30_000
+    });
+
+    const message = {
+      typeUrl: `/akash.deployment.v1beta3.MsgCreateDeployment`,
```

Comment:
messages are kept in https://github.com/akash-network/console/blob/main/apps/api/src/billing/services/rpc-message-service/rpc-message.service.ts

Reply:
@jzsfkzm Just this and lgtm!